### PR TITLE
Filing a future against copy initializers with default args

### PR DESCRIPTION
**Pull request by bradcray**
This pair of tests demonstrates that if an initializer meets
the copy initializer requirements, but also has additional
arguments, it is not considered.  The second test shows an
unfortunate workaround that can be used in which a second
copy initializer is added to work around the issue.

### CODE REVIEW

**Comment by bradcray** [comment]
@lydia-duncan and @mppf: This looks like it's a bug to me, but I wanted to get one or both of you to verify before merging it and opening an issue against it.  Thanks!

**Comment by bradcray** [comment]
(This was motivated by trying to convert the `string` constructor to an initializer)

File: test/types/records/init/copy/copyWithDefaultArgs.chpl [hunk 1]
```diff
@@ -0,0 +1,31 @@
+record R {
+  var x: int;
+
+  proc init(x: int, y: bool = true) {
+    if y then
+      this.x = x;
+  }
+
+  proc init(r: R, y: bool = true) {
```
Comment (mppf) [inline test/types/records/init/copy/copyWithDefaultArgs.chpl]: It seems odd to me that there are two duplicate copies of `init` here - wouldn't that lead to ambiguity if everything was "working" ?

**Comment by mppf** [comment]
I've also had trouble when I tried to use copy initializers in practice but I'm not sure what's going wrong.

File: test/types/records/init/copy/copyWithDefaultArgs.chpl (hunk 1 above)
Comment (lydia-duncan) [reply test/types/records/init/copy/copyWithDefaultArgs.chpl]: Not for these two, you can distinguish them based on the argument without a default value (one only takes an int for the first argument, the other only takes an instance of R for the first argument)

Comment (mppf) [reply test/types/records/init/copy/copyWithDefaultArgs.chpl]: Ah, I missed that for some reason. Thanks.

**Comment by lydia-duncan** [comment]
I agree that this is a bug on copy initializers, they should allow an initializer where all additional arguments have a default value.

**Pull request merged**